components/tor/tor_control.h

```diff
@@ -155,15 +138,7 @@ class TorControl {
 
   TorControl::Delegate* delegate_;
```

Comment:
IMO it is more descriptive to explicitly hold `WeakPtr` to the delegate here, but for now we can leave it alone

Reply:
actually I've just understood that normally we shouldn't hold a raw pointer because if the pointee object dies, we are going to crash when trying to call `AsWeakPtr()`